### ISSUE TITLE
%(namelower)s fails to expand in Python setuptools download with --try-toolchain

### ISSUE THREAD

**Issue by zao** [easybuilders/easybuild-framework#2618]
Using the patch from #2617 to handle the problem in #2615, the following failure to download setuptools occurred when running `eb Python-2.7.15-foss-2018b.eb -r --try-toolchain=foss,2018.08 --debug`.

As this is the first time I ever run develop, I have no idea if it affects regular un-try builds, and if the patch had this effect.
```
ERROR: Build of /tmp/eb-xeEGWZ/tweaked_easyconfigs/Python-2.7.15-foss-2018.08.eb failed (err: "build failed (first 300 chars): Couldn't find file setuptools-39.2.0.zip anywhere, and downloading it didn't work either... Paths attempted (in order): /tmp/eb-xeEGWZ/tweaked_easyconfigs/p/Python/extensions/setuptools-39.2.0.zip, /tmp/eb-xeEGWZ/tweaked_easyconfigs/p/Python/packages/setuptools-39.2.0.zip, /tmp/eb-xeEGWZ/tweaked_eas")
```
```
== 2018-10-12 20:58:17,199 filetools.py:1212 DEBUG Not creating existing path /home/zao/.local/eb-develop/sources/p/Python
== 2018-10-12 20:58:17,199 easyblock.py:728 DEBUG Trying to download file setuptools-39.2.0.zip from https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip ...
== 2018-10-12 20:58:17,199 filetools.py:484 DEBUG Trying to download setuptools-39.2.0.zip from https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip
== 2018-10-12 20:58:17,199 filetools.py:491 DEBUG Using timeout of 10 seconds for initiating download
== 2018-10-12 20:58:17,199 filetools.py:1184 INFO Creating directory /home/zao/.local/eb-develop/sources/p/Python/extensions (parents: True, set_gid: False, sticky: False)
== 2018-10-12 20:58:17,257 filetools.py:535 WARNING IOError occurred while trying to download https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip: <urlopen error [Errno -
2] Name or service not known>
== 2018-10-12 20:58:17,258 filetools.py:549 INFO Attempt 1 of downloading https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, trying again...
== 2018-10-12 20:58:17,259 filetools.py:535 WARNING IOError occurred while trying to download https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip: <urlopen error [Errno -2] Name or service not known>
== 2018-10-12 20:58:17,259 filetools.py:549 INFO Attempt 2 of downloading https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, trying again...
== 2018-10-12 20:58:17,260 filetools.py:535 WARNING IOError occurred while trying to download https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip: <urlopen error [Errno -2] Name or service not known>
== 2018-10-12 20:58:17,260 filetools.py:555 WARNING Download of https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, done trying
== 2018-10-12 20:58:17,260 easyblock.py:728 DEBUG Trying to download file setuptools-39.2.0.zip from http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip ...
== 2018-10-12 20:58:17,260 filetools.py:484 DEBUG Trying to download setuptools-39.2.0.zip from http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip
== 2018-10-12 20:58:17,261 filetools.py:491 DEBUG Using timeout of 10 seconds for initiating download
== 2018-10-12 20:58:17,261 filetools.py:1212 DEBUG Not creating existing path /home/zao/.local/eb-develop/sources/p/Python/extensions
== 2018-10-12 20:58:17,471 filetools.py:529 WARNING URL http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip was not found (HTTP response code 404), not trying again
== 2018-10-12 20:58:17,471 filetools.py:555 WARNING Download of http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, done trying
== 2018-10-12 20:58:17,496 build_log.py:158 ERROR EasyBuild crashed with an error (at ?:124 in __init__): Couldn't find file setuptools-39.2.0.zip anywhere, and downloading it didn't work either...
```

**Comment by zao** [comment]
Expansion succeeds when building `eb Python-2.7.15-foss-2018b.eb -r --debug`, that is, without `--try-toolchain`.

**Comment by zao** [comment]
This seems to occur even without the patch, so it's not the root cause, just develop being develop?

**Comment by ocaisa** [comment]
I think this is just a general problem with the easyconfig `dump()` method (that is now also being used with `--try-toolchain`). It tries to be a little bit too clever and replaces every instance of `python` with the template value  `%(namelower)s`, but in the case of extensions this is not the correct thing to do and just breaks the URL.

**Comment by ocaisa** [comment]
@boegel You might be able to fix this faster than me, I don't know that part of the code very well.

**Comment by zao** [comment]
@ocaisa Oh, you're right... I only ever looked at the debug log, never at the Python easyconfig itself where there's indeed literal `pypi.python.org` URLs :)

**Comment by pescobar** [comment]
I am also experiencing this issue

**Comment by ocaisa** [comment]
@pescobar There's a verified fix in #2619, but I can't seem to come up with a working test for it

**Comment by boegel** [comment]
closing this now that #2619 is merged
